components/match2/wikis/starcraft2/legacy/match_maps_team_legacy.lua

```diff
@@ -0,0 +1,182 @@
+			and storageArgs[prefix .. 'p' .. side .. 'link']
+			or storageArgs[prefix .. 'p' .. side],
+		['t' .. side .. 'p1race'] = storageArgs[prefix .. 'p' .. side .. 'race'],
+		['t' .. side .. 'p1flag'] = storageArgs[prefix .. 'p' .. side .. 'flag'],
+
+		['opponent' .. side .. 'archon'] = archon and 'true' or nil,
+		['opponent' .. side .. 'race'] = archon and storageArgs[prefix .. 'p' .. side .. 'race'] or nil,
+
+		['t' .. side .. 'p2'] = String.isNotEmpty(storageArgs[prefix .. 't' .. side .. 'p2link'])
+			and storageArgs[prefix .. 't' .. side .. 'p2link']
+			or storageArgs[prefix .. 't' .. side .. 'p2'],
+		['t' .. side .. 'p2race'] = storageArgs[prefix .. 't' .. side .. 'p2race'],
+		['t' .. side .. 'p2flag'] = storageArgs[prefix .. 't' .. side .. 'p2flag'],
+
+		['t' .. side .. 'p3'] = String.isNotEmpty(storageArgs[prefix .. 't' .. side .. 'p3link'])
+			and storageArgs[prefix .. 't' .. side .. 'p3link']
+			or storageArgs[prefix .. 't' .. side .. 'p3'],
+		['t' .. side .. 'p3race'] = storageArgs[prefix .. 't' .. side .. 'p3race'],
+		['t' .. side .. 'p3flag'] = storageArgs[prefix .. 't' .. side .. 'p3flag'],
+	}
+
+	MatchMapsTeamLegacy._setPlayersForOpponents(addToMapArgs, side, storageArgs[prefix .. 'p' .. side])
+
+	return Table.mergeInto(mapArgs, addToMapArgs)
+end
+
+function MatchMapsTeamLegacy._setPlayersForOpponents(args, side, displayName)
+	local prefix = 't' .. side .. 'p'
+
+	for playerKey, player in Table.iter.pairsByPrefix(args, prefix) do
+		_opponentPlayers[side][player] = {
+			race = args[playerKey .. 'race'],
+			flag = args[playerKey .. 'flag'],
+			display = displayName,
+		}
+	end
+end
+
+function MatchMapsTeamLegacy._removeProcessedMapInput(prefix)
+	for key, _ in pairs(_storageArgs) do
+		if String.startsWith(key, prefix) then
+			_storageArgs[key] = nil
+		end
+	end
+end
+
+function MatchMapsTeamLegacy._handleOpponents()
+	local storageArgs = _storageArgs
+	local args = _args
+
+	for opponentIndex = 1, _NUMBER_OF_OPPONENTS do
+		if args['team' .. opponentIndex] and args['team' .. opponentIndex]:lower() == 'bye' then
+			storageArgs['opponent' .. opponentIndex] = {
+				['type'] = 'literal',
+				name = 'BYE',
+			}
+		else
+			storageArgs['opponent' .. opponentIndex] = {
+				['type'] = 'team',
+				template = args['team' .. opponentIndex],
+				score = args['score' .. opponentIndex],
+			}
+			if args['team' .. opponentIndex] == '' then
+				storageArgs['opponent' .. opponentIndex]['type'] = 'literal'
+			else
+				local players = {}
+				local playerIndex = 1
+				for player, playerData in pairs(_opponentPlayers[opponentIndex]) do
+					players['p' .. playerIndex .. 'link'] = player
+					players['p' .. playerIndex] = playerData.display
+					players['p' .. playerIndex .. 'flag'] = playerData.flag
+					players['p' .. playerIndex .. 'race'] = playerData.race
+					playerIndex = playerIndex + 1
+				end
+				storageArgs['opponent' .. opponentIndex].players = players
```

Comment:
This would change `_storageArgs` as well no? Is that intended? Happens a lot throughout this file

Reply:
yeah that is the intention
i can rename all the `storageArgs` to `_storageArgs` if you want i was just lazy to always write the `_` hence used `local storageArgs = _storageArgs` at the start of the functions

Reply:
Yea please use `_storageArgs`, it's confusing otherwise